api/src/main/java/au/org/aodn/nrmn/restapi/service/SurveyEditService.java

```diff
@@ -0,0 +1,150 @@
+                    "The survey time must be in the format hh:mm[:ss]"));
+        }
+
+        // Site code and site name exist and match
+        if (!surveyDtoSite.getSiteName().equals(surveyDto.getSiteName())) {
+            errors.add(new ValidationError("Site", "siteCode", surveyDto.getSiteName(),
+                    String.format("The site code %s does not match the site name \"%s\"", surveyDto.getSiteCode(), surveyDto.getSiteName())));
+        }
+
+        // Lat and Lon match site
+        String lat = surveyDto.getLatitude();
+        if(!isEmpty(lat) && !truncateNumber(surveyDtoSite.getLatitude()).equals(truncateNumber(Double.parseDouble(lat)))){
+            errors.add(new ValidationError("Survey", "latitude", surveyDto.getSiteName(),
+                    String.format("The survey latitude does not match the latitude of the site (%s)", surveyDtoSite.getLatitude())));
+        }
+
+        String lon = surveyDto.getLongitude();
+        if(!isEmpty(lon) && !truncateNumber(surveyDtoSite.getLongitude()).equals(truncateNumber(Double.parseDouble(lon)))){
+            errors.add(new ValidationError("Survey", "longitude", surveyDto.getSiteName(),
+                    String.format("The survey longitude does not match the longitude of the site (%s)", surveyDtoSite.getLongitude())));
+        }
+
+        // Direction Validation
+        if (!EnumUtils.isValidEnum(Directions.class, surveyDto.getDirection())) {
+            errors.add(new ValidationError("Survey", "direction", surveyDto.getDirection(),
+                    surveyDto.getDirection() + " is invalid, expected: N,NE,E,SE,S,SW,W,NW"));
+        }
+
+        // Ensure site/date/depth.surveyNum is unique
+        Integer depth = surveyDto.getDepth() != null && surveyDto.getDepth().length() > 0 ? Integer.valueOf(surveyDto.getDepth()) : null;
+        List<Survey> duplicateSurveys = surveyRepository.findBySiteDepthSurveyNumDate(
+                surveyDtoSite, depth, surveyDto.getSurveyNum(), surveyDate).stream()
+                .filter(s -> !s.getSurveyId().equals(surveyDto.getSurveyId()))
+                .collect(Collectors.toList());
+
+        if (duplicateSurveys.size() > 0) {
+            errors.add(new ValidationError("Survey", "surveyNum", surveyDto.getSurveyNum().toString(),
+                    String.format("A survey with the site, date and depth of \"%s/%s/%s.%s\" already exists.",
+                            surveyDto.getSiteCode(), surveyDto.getSurveyDate(), surveyDto.getDepth(), surveyDto.getSurveyNum())));
+        }
+
+        return new ValidationErrors(errors);
+    }
+
+    private String truncateNumber(Double number) {
+
+        if(number == null) {
+            return null;
+        }
+
+        DecimalFormat df = new DecimalFormat("#.####");
+        df.setRoundingMode(RoundingMode.HALF_UP);
+        return String.valueOf(Double.parseDouble(df.format(number)));
+    }
+
+    private boolean isEmpty(String val){
```

Comment:
Probably use StringUtils.isNull instead.

Reply:
Ah! Perfect fit, I didn't know about that util.